File: backend/src/main/java/fr/pedalons/dto/social/response/SocialIdentityDto.java
package fr.pedalons.dto.social.response;

import fr.pedalons.domain.social.UserSocialIdentity;
import fr.pedalons.dto.validation.ValidateSchema;
import fr.pedalons.enums.SocialProvider;
import java.time.Instant;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "A linked external identity (e.g. Strava)")
@ValidateSchema
public record SocialIdentityDto(
    @Schema(description = "Provider identifier", required = true) SocialProvider provider,
    @Schema(description = "Display name of the provider", required = true) String displayName,
    @Schema(description = "When the identity was linked", required = true) Instant linkedAt) {

  public static SocialIdentityDto from(UserSocialIdentity identity) {
    return new SocialIdentityDto(
        identity.getProvider(), identity.getProvider().getDisplayName(), identity.getCreatedAt());
  }
}

File: frontend/src/api/dto/socialIdentityDto.ts
import type { Instant } from './instant.ts'
import type { SocialProvider } from './socialProvider.ts'

/**
 * A linked external identity (e.g. Strava)
 */
export interface SocialIdentityDto {
  /** Provider identifier */
  provider: SocialProvider
  /** Display name of the provider */
  displayName: string
  /** When the identity was linked */
  linkedAt: Instant
}
